resources/js/features/game-list/hooks/useTableSync.ts

```diff
@@ -1,25 +1,61 @@
-import type { ColumnFiltersState, PaginationState, SortingState } from '@tanstack/react-table';
-import { useUpdateEffect } from 'react-use';
+import type {
+  ColumnFiltersState,
+  PaginationState,
+  SortingState,
+  TableState,
+  VisibilityState,
+} from '@tanstack/react-table';
+import { useCookie, useUpdateEffect } from 'react-use';
+
+import { usePageProps } from '@/common/hooks/usePageProps';
 
 interface UseAutoUpdatingQueryParamsProps {
   columnFilters: ColumnFiltersState;
+  columnVisibility: VisibilityState;
   defaultFilters: ColumnFiltersState;
   pagination: PaginationState;
   sorting: SortingState;
 
   defaultPageSize?: number;
+  isUserPersistenceEnabled?: boolean;
 }
 
 /**
- * This hook is designed to keep the URL query params in sync with the table state.
+ * This hook is designed to keep the URL query params and
+ * user's persistence cookie in sync with the table state.
  */
-export function useAutoUpdatingQueryParams({
+export function useTableSync({
   columnFilters,
+  columnVisibility,
   pagination,
   sorting,
   defaultFilters = [],
   defaultPageSize = 25,
+  isUserPersistenceEnabled = false,
 }: UseAutoUpdatingQueryParamsProps) {
+  const { persistenceCookieName } = usePageProps<{ persistenceCookieName: string }>();
+
+  const [cookie, setCookie, deleteCookie] = useCookie(persistenceCookieName);
+
+  useUpdateEffect(() => {
+    if (isUserPersistenceEnabled) {
+      // Don't persist filtering by title.
+      const persistedFilters = columnFilters.filter((filter) => filter.id !== 'title');
```

Comment:
This doesn't seem to be working. The value is still appearing in my cookie:
```
%22sorting%22:[{%22id%22:%22title%22%2C%22desc%22:false}]
```

Reply:
The cookie value demonstrated (without encoding) is:
```
"sorting": [{"id":"title","desc":false}]
```
The logic on this line is for ensuring that a filter by `title` is not persisted in the cookie. For example, if you search for "zelda" on the All Games list, then navigate directly to `/games`, the search (title filter) should not persist.